week2/project/ecommerce-app/src/components/Products.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState } from 'react'
+import { NavLink } from 'react-router-dom'
+// import {Button} from 'bootstrap'
+
+
+function Products() {
+    const [data, setData] = useState([])
+    const [filter, setFilter] = useState(null)
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const response = await fetch('https://fakestoreapi.com/products')
+                if (!response.ok) {
+                    throw new Error('Network response was not ok!')
+                }
+                const jsonData = await response.clone().json()
+                // console.log("Fetched data:", jsonData);
+                setData(jsonData)
+                setLoading(false)
+            } catch (error) {
+                setError(error)
+                setLoading(false)
+            }
+        }
+        if (loading) {
+            fetchData()
+        }
+    }, [loading])
+
+
+    const Loading = () => {
+        if (loading) {
+            return <div>Loading... </div>
+        }
+
+        if (error) {
+            return <div>Error: {error.message}</div>
+        }
+    }
```

Comment:
Great work, making a separate component for showing loading! 💪 

A tip: I see this exact same code In two different files. Maybe you can create a separate Loading and Error component In the **components** folder and render this? Example:

```suggestion
        if (loading) {
            return <Loader /> // this Loader component renders Loading...
        }
```